### PR TITLE
feat: support overviewHeaders and subPage overview

### PR DESCRIPTION
**Pull request by Timeless0911**
## Summary

1. support `overviewHeaders` to control the minimum title displayed in the overview page, config in `_meta.json` and front matter are both takes effect.
```json
[
  {
    "type": "file",
    "name": "react-apis",
    "overviewHeaders": [2]
  },
  {
    "type": "dir",
    "name": "ui-props",
    "label": "css.ui-props",
    "overviewDepth": [],
    "collapsible": true,
    "collapsed": true
  },
]
```

```mdx
---
overviewHeaders: [2]
---
```

2. support subpage overview by writing a file with the same name as sidebar in the upper folder of sidebar with frontmatter
```mdx
---
overview: true
---
```
image

Or writing a `index.mdx` in the siderbar dir folder with frontmatter and register it in `_meta.json`

image

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [x] I have updated the documentation.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: .changeset/loud-cameras-occur.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@rspress/plugin-auto-nav-sidebar": minor
+"@rspress/theme-default": minor
+"@rspress/shared": minor
+---
+
+support overviewDepth to control the minimum title displayed in the overview page
```
Comment (chenjiahan) [inline .changeset/loud-cameras-occur.md:7]: Can we have some E2E cases for this new feature?

Comment (Timeless0911) [reply .changeset/loud-cameras-occur.md:7]: done

**Comment by Timeless0911** [comment]
!canary

**Comment by sanyuan0704** [comment]
Is `overviewHeaders` instead `overviewDepth` a better choice? We can support the following config:

```js
{
  "overviewHeaders": [2, 3]
}
```

**Comment by sanyuan0704** [comment]
The overview config may be supported in frontmatter:

```md
---
overviewHeaders: [2]
---
```

Refer `extractApiHeaders` in vuepress https://vt.insx.dev/api/config-frontmatter.html#extractapiheaders

**Comment by Timeless0911** [comment]
!canary

**Comment by sanyuan0704** [comment]
🎉

**Review by sanyuan0704 — APPROVED** [review]

**Comment by chenjiahan** [comment]
👏

**Pull request merged**